### ISSUE TITLE
Getting "Validation failed in API: certs failed to find cert that matched private key" for certain projects

### ISSUE THREAD

**Issue by proligde** [rancher/rancher#24020]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
we were not able to reproduce the problem - nor to solve it

**Result:**
Randomly (after weeks of working properly) one of our project generates an Error when accessing it: `Validation failed in API: certs failed to find cert that matched private key`. We were unable to fix the problem and had to delete and recreate the project to get rid of the error.

Some weeks later another Project now has the same problem.

**Other details that may be helpful:**

We tried to investigate the problem and found out that:
- Manually rotating keys didn't help
- Other projects within the same cluster are NOT affected
- Accessing the workloads within the affected project using kubectl works flawlessly - all namespaces, workloads and so on are accessible - workloads are still working properly. It seems to be a problem solely related to rancher's own "projects" concept.
- This error occurrs when accessing the project's workloads via the Rancher GUI or the API for one of the projects "namespaced*" services - for example:

`https:/rancher-host/v3/projects/c-<id>:p-<id>/namespacedbasicauths`

`{
"baseType": "error",
"code": "InvalidBodyContent",
"fieldName": "certs",
"message": "failed to find cert that matched private key",
"status": 422,
"type": "error"
}`

- The Problem is server related since every user in every browser (even in incognito mode) now has the same problem.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v.2.3.1 (but occurred in 2.9.x already)
- Installation option (single install/HA): single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom - created by Rancher
- Machine type (cloud/VM/metal) and specifications (CPU/memory): Cloud / several nodes affected
- Kubernetes version (use `kubectl version`): v1.15.4 (occurred in earlier versions as well)

- Docker version (use `docker version`): several docker versions affected

**Comment by proligde** [comment]
I might add: We don't care so much why this happens every few weeks (while it shouldn't, obviously) - we just need a way to fix it, other than to delete and recreate the whole project. We're at a point where this could be a showstopper for using Rancher as our K8S-Maintenance tool although we really love it.

**Comment by proligde** [comment]
We were finally able to fix the problem:

We assumed that this error had something to do with the inter-node-communication between the k8s nodes and/or rancher. This now seemed not to be the case:

We found additional log messages on local worker nodes like

`
Nov 13 12:09:50 my-hostname 7b73c950acf8[2161]: I1113 11:09:50.964852       9 store.go:387] secret my-namespace-name/my-service-name.local-tls was updated and it is used in ingress annotations. Parsing...
Nov 13 12:09:50 my-hostname 7b73c950acf8[2161]: W1113 11:09:50.966579       9 backend_ssl.go:48] Error obtaining X.509 certificate: unexpected error creating pem file: tls: private key does not match public key
`

which pointed to a certain workload within our failing project. Using `kubectl` we first removed the ingress and then we also removed the secrets/certificates previously generated by cert-manager (acme/letsencrypt). We assume that something broke parts of the certificate retrieved by cert-manager (we are using `jetstack-cert-manager-controller:v0.9.1` at the moment).

I hope this helps anyone who came across the same problem.
